File: fleet_platform/schemas/monitoring.py
"""Monitoring API response schemas."""
from __future__ import annotations

from pydantic import BaseModel


class NodeCountsSchema(BaseModel):
    online: int
    stale: int
    offline: int
    unknown: int
    total: int


class AlertEventSchema(BaseModel):
    id: str
    message: str
    fired_at: str | None


class CeleryQueuesSchema(BaseModel):
    default: int
    maintenance: int
    drift: int
    sbom: int
    active: int


class HttpRequestSchema(BaseModel):
    handler: str
    method: str
    status_code: str
    count: int


class FleetHealthSchema(BaseModel):
    node_count: int
    avg_cpu_load_1m: float | None
    avg_mem_used_pct: float | None
    avg_disk_pct: float | None
    thermal_ok: int | None
    nodes_with_gpu: int
    total_gpu_vram_mb: int


class MaintenanceHeartbeatSchema(BaseModel):
    last_run_at: str | None
    age_seconds: int | None
    beat_ok: bool | None   # None = Redis unavailable


class MonitoringSummarySchema(BaseModel):
    node_counts: NodeCountsSchema
    alert_events_24h: list[AlertEventSchema]
    alert_count_24h: int
    celery_queues: CeleryQueuesSchema
    http_requests: list[HttpRequestSchema]
    fleet_health: FleetHealthSchema
    maintenance_heartbeat: MaintenanceHeartbeatSchema
    generated_at: str

File: frontend/src/api/monitoring.ts
import { api } from './client'

export interface NodeCounts {
  online: number
  stale: number
  offline: number
  unknown: number
  total: number
}

export interface AlertEvent {
  id: string
  message: string
  fired_at: string | null
}

export interface CeleryQueues {
  default: number
  maintenance: number
  drift: number
  sbom: number
  active: number
}

export interface HttpRequest {
  handler: string
  method: string
  status_code: string
  count: number
}

export interface FleetHealth {
  node_count: number
  avg_cpu_load_1m: number | null
  avg_mem_used_pct: number | null
  avg_disk_pct: number | null
  thermal_ok: number | null
  nodes_with_gpu: number
  total_gpu_vram_mb: number
}

export interface MonitoringSummary {
  node_counts: NodeCounts
  alert_events_24h: AlertEvent[]
  alert_count_24h: number
  celery_queues: CeleryQueues
  http_requests: HttpRequest[]
  fleet_health: FleetHealth
  generated_at: string
}

export const monitoringApi = {
  getSummary: () => api.get<MonitoringSummary>('/api/v1/monitoring/summary'),
}
